### ISSUE TITLE
Compare/Diff appears to be the wrong way around

### ISSUE THREAD

**Issue by PromoFaux** [arnoudkooi/ServiceNow-Utils#370]
Hi there - great product, it makes a lot of my taks much easier!

I've just stumbled across the compare functionality and this button when comparing an update set:

![image](https://user-images.githubusercontent.com/1998970/221577318-fdc5d9f3-280d-49b0-b31c-6a1319721a42.png)

When comparing changes in update sets, I would expect the _existing_ version to be on the left, and the changes on the right. Looking at the 

![image](https://user-images.githubusercontent.com/1998970/221577084-8b6ffc2c-0de5-4d3c-baf2-e88f70cde3f9.png)

Not sure if this is down to how the change is displayed initially on ServiceNow (i.e new on the left, old on the right), or if it can be worked around on your side? Not a big deal, just used to seeing new changes in green ;)

Thanks for your work on this extension!

**Comment by arnoudkooi** [comment]
For me the current way makes more sense, newest on the right...

**Comment by PromoFaux** [comment]
Yes, that is what I am saying... 

Newest _should_ be on the right... in my screenshot above, the newest is showing on the left

**Comment by jacebenson** [comment]
I would love somehow note the updateset/family/package whatever on both sides.

**Comment by arnoudkooi** [comment]
I just tested, for me the newest is on the right.. Can you share steps to reproduce

**Comment by PromoFaux** [comment]
Sure - here is a gif - forgive the clumsy obfuscation, just making sure there are no company details in there

![diff](https://user-images.githubusercontent.com/1998970/221643091-1452aba1-e818-45e0-b162-6546a3bd942a.gif)

**Comment by arnoudkooi** [comment]
That is very hard to follow and doesn't give me additional info to confirm the issue

**Comment by PromoFaux** [comment]
Apologies, I figured it would be easier to follow. In a previewed update set:

 - Click on one of the updates contained therein
 - Click "compare to current" related link
 - scroll to the part that is changed
 - click the snutils compare hyperlink
 - observe the diff - noting that they are the wrong way around.

I'm not sure what additional information I can give you here - or how much clearer I can make it 😏

I'm Happy to admit if I'm just using the tool wrong - but I can definitely reproduce this multiple times across different update sets.

> I just tested, for me the newest is on the right.

What steps did you take to test?

**Comment by eastolbe** [comment]
The GIF shows a non-committed update set, so Current is the existing record, and the Selected is the new update from the update set. I guess typically Current would be the most recent update so should always show on the right, but in this use case, Current isn't the latest. I think this is the only scenario where the more recent update isn't shown on the right side in the diff screen.

**Comment by PromoFaux** [comment]
I think I'm very confused here... And like I said, maybe I'm using the tool in the wrong way. It just seems like if you wanted to review what an incoming update was going to do to the current, you would want the update on the right.

I'm new to servicenow, so maybe things are done differently in this world 🙃

Maybe I have stumbled across a an edgecase here - but would it be too much to ask for a "swap" button on the diff output itself?

**Comment by VanWeapon** [comment]
I'm with Adam here, newest should be on the right but this isn't SN Utils
fault, for uncommitted update sets the in-platform diff checker puts
current on the right and new (remote) on the left, it's very confusing

+1 for a swap button because I think trying to hard code which places to
swap sides and which places to leave as is would be problematic

**Comment by jacebenson** [comment]
Would be a lot easier to understand what's what if the editors could somehow be labeld by update set/scope/date/etc.

**Comment by crysallis** [comment]
I was just about to say something similar... if you "Preview" an update set... imported or loaded after a clone etc... Select an update entry and Compare to Current the "Current Version" is the file as it is already on platform... not as it will be after the update set is applied... The left is the "Selected **Update**" and will overwrite the "Current Version" if applied.  I think the + - are the confusing bits... as they are implying that I'm adding something I'm not... when it's really just showing what's IN the file... vs what's NOT IN the file... the left version is always what will be if you revert or commit. 

compare to current using the history... you are looking at the "Selected Version" vs the "Selected Update" but the same view is there.

**Comment by arnoudkooi** [comment]
Interesting discussion, I was just about to implement it like this in [compare.js](https://github.com/arnoudkooi/ServiceNow-Utils/blob/a0a0240b90461f3f7c2e15b8f2b4e45cbeaef6fc/js/monaco/compare.js)

However, maybe adding a clarifying header is a better idea. I'll leave it for now.

```javascript
        let data = {
            fieldName : fieldName,
            leftBody: leftBody,
            leftTitle: names.leftTitle,
            rightBody: rightBody,
            rightTitle: names.rightTitle,
            snusettings: snusettings
        };

        //In case of uncomitted update show current version in left pane (Issue #370)
        if (location.pathname == '/merge_form_current_update.do'){
            data = {
                leftBody: rightBody,
                leftTitle: names.rightTitle,
                rightBody: leftBody,
                rightTitle: names.leftTitle,
            };
        }
```

**Comment by arnoudkooi** [comment]
For now apply the above solution, will look at further improvements later.

**Comment by arnoudkooi** [comment]
Please stand by for 6.3.6.6 as 6.3.6.5 broke the diff viewer for updates.

**Comment by PromoFaux** [comment]
Thanks for the implementation, confirmed it is working as expected for me currently:

![image](https://user-images.githubusercontent.com/1998970/222403497-334ba274-7aaf-4ec2-9398-473877e9db18.png)

**Comment by arnoudkooi** [comment]
Good, closing this one. Will keep the suggested improvements in mind.
